TheMovieDb/Components/Movies Section/CategoryTableViewCell.swift

```diff
@@ -0,0 +1,185 @@
+  override func awakeFromNib() {
+    super.awakeFromNib()
+    setUpUI()
+    setupCollectionView()
+    requestAPI()
+    // Initialization code
+//    DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
+//      self.collectionView.reloadData()
+//    }
+  }
+  
+  // Request for trending movies
+  private func requestAPITrending() {
+    let completion: (Result<MovieList, Error>) -> Void = { [weak self] result in
+        debugPrint(result)
+        switch result {
+        case .success(let response):
+          self?.trendingMovies = response.results
+          print(response)
+          self?.collectionView.reloadData()
+          
+        default:
+          self?.trendingMovies = []
+        }
+    }
+    API.getTrendingMovies.resume(completion: completion)
+  }
+  
+  private func requestAPINowPlaying() {
+    let completion: (Result<MovieList, Error>) -> Void = { [weak self] result in
+        debugPrint(result)
+        switch result {
+        case .success(let response):
+          self?.nowPlayingMovies = response.results
+          print(response)
+          self?.collectionView.reloadData()
+          
+        default:
+          self?.nowPlayingMovies = []
+        }
+    }
+    API.getNowPlayingMovies.resume(completion: completion)
+  }
+  
+  private func requestAPIPopular() {
+    let completion: (Result<MovieList, Error>) -> Void = { [weak self] result in
+        debugPrint(result)
+        switch result {
+        case .success(let response):
+          self?.popularMovies = response.results
+          print(response)
+          self?.collectionView.reloadData()
+          
+        default:
+          self?.popularMovies = []
+        }
+    }
+    API.getPopularMovies.resume(completion: completion)
+  }
+  
+  private func requestAPITopRated() {
+    let completion: (Result<MovieList, Error>) -> Void = { [weak self] result in
+        debugPrint(result)
+        switch result {
+        case .success(let response):
+          self?.topRatedMovies = response.results
+          print(response)
+          self?.collectionView.reloadData()
+          
+        default:
+          self?.topRatedMovies = []
+        }
+    }
+    API.getTopRatedMovies.resume(completion: completion)
+  }
+  
+  private func requestAPIUpcoming() {
+    let completion: (Result<MovieList, Error>) -> Void = { [weak self] result in
+        debugPrint(result)
+        switch result {
+        case .success(let response):
+          print("Upcoming \(response)")
+          self?.upcomingMovies = response.results
+          print(response)
+          self?.collectionView.reloadData()
+          
+        default:
+          print("Upcoming not found")
+          self?.upcomingMovies = []
+        }
+    }
+    API.getUpcomingMovies.resume(completion: completion)
+  }
+  
+  private func requestAPI() {
+    requestAPITrending()
+    requestAPIPopular()
+    requestAPIUpcoming()
+    requestAPITopRated()
+    requestAPINowPlaying()
+  }
```

Comment:
this cell implement a lot of logic that could be moved, use Single Responsibility Principle